### PR TITLE
Update clap to v4

### PR DESCRIPTION
**Pull request by dnaka91**
Updating the `clap` crate to the just released v4.0. Am currently updating many of my own crates to v4 and found issue #496 as part of the Hacktoberfest event.

I tried to keep the implementation as much as-is as possible, but especially clap's help message styling changed a lot with v4.

You might wonder the `AppSettings::DeriveDisplayOrder` being removed, but it is not needed anymore, as this is the new default now. Besides that, I used the `ValueEnum` derive a lot to have the possible values of enums provided to clap. This may have _invaded_ some modules with clap use statements, so just let me know if you don't want that, and I'll think of something else.

This is my first contribution to this project, so I just wanted to say: Thank you for this awesome project, I'm using it regularly for my project and love it 🚀.

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -28,6 +28,7 @@ pub enum OutputFormat {
     /// Minimal output, usually just the result, such as the MSRV or whether verify succeeded or failed
     Minimal,
     /// No output -- meant to be used for debugging and testing
+    #[value(hide = true)]
```
Comment (foresterre) [inline src/config.rs:31]: What happens when `hide = true`? Will it not show up in the help (i.e. hidden), or be disabled as a value altogether? 

This value would previously only be set when the `--no-user-output` was given; if it is hidden, not disabled, would it be possible to disable it, so it's not available as `--user-output none`.

**Review by foresterre — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: src/config.rs (hunk 1 above)
Comment (dnaka91) [reply src/config.rs:31]: This setting will only hide it from help and shell completions ([docs](https://docs.rs/clap/latest/clap/builder/struct.PossibleValue.html#method.hide)). That means it's not visible, but still valid to do `--user-output none`.

I've looked around and didn't find any setting that would fully remove this variant from the `ValueEnum` implementation. Would have hoped for something like `#[value(skip)]`, but there isn't.

Currently, the only way seems to be to implement `ValueEnum` by hand. But that's rather easy, so I'll go with that, to keep the original behavior.

Comment (dnaka91) [reply src/config.rs]: Oh, I was wrong, there is actually `#[value(skip)]`, but it's not mentioned anywhere in the docs. It's only mentioned in regard to subcommands.

I pushed an update right now, so it'll not be part of the valid arguments anymore. I verified that it's actually excluded, by utilizing `cargo-expand`. Just to make sure, as it's not mentioned in the docs.

**Review by foresterre — APPROVED** [review]

**Comment by foresterre** [comment]
Thank you for the changes!

**Pull request merged**